gaspy/atoms_operators.py

```diff
@@ -251,6 +252,172 @@ def find_adsorption_sites(atoms):
     sites = sites_dict['all']
     return sites
 
+def find_bulk_cn_dict(bulk_atoms):
+    '''
+    Get a dictionary of coordination numbers
+    for each distinct site in the bulk structure.
+
+    Taken from pymatgen.core.surface Class Slab
+    `get_surface_sites`.
+    https://pymatgen.org/pymatgen.core.surface.html
+    '''
+    struct = AseAtomsAdaptor.get_structure(bulk_atoms)
+    sga = SpacegroupAnalyzer(struct)
+    sym_struct = sga.get_symmetrized_structure()
+    unique_indices = [equ[0] for equ in sym_struct.equivalent_indices]
+    # Get a dictionary of unique coordination numbers
+    # for atoms in each structure.
+    # for example, Pt[1,1,1] would have cn=3 and cn=12
+    # depends on the Pt atom.
+    v = VoronoiNN()
+    cn_dict = {}
+    for idx in unique_indices:
+        elem = sym_struct[idx].species_string
+        if elem not in cn_dict.keys():
+            cn_dict[elem] = []
+        cn = v.get_cn(sym_struct, idx, use_weights=True)
+        cn = float('%.5f' %(round(cn, 5)))
+        if cn not in cn_dict[elem]:
+            cn_dict[elem].append(cn)
+    return cn_dict
+
+
+def find_surface_atoms_indices(bulk_cn_dict, atoms):
+    '''
+    A helper function referencing codes from pymatgen to
+    get a list of surface atoms indices of a slab's
+    top surface. Due to how our workflow is setup, the
+    pymatgen method cannot be directly applied.
+
+    Taken from pymatgen.core.surface Class Slab,
+    `get_surface_sites`.
+    https://pymatgen.org/pymatgen.core.surface.html
+
+    Arg:
+        bulk_cn_dict    A dictionary of coordination numbers
+                        for each distinct site in the respective bulk structure
+        atoms           The slab where you are trying to find surface sites in
+                        `ase.Atoms` format
+    Output:
+        indices_list    A list that contains the indices of
+                        the surface atoms
+    '''
+    struct = AseAtomsAdaptor.get_structure(atoms)
+    v = VoronoiNN()
```

Comment:
Single-letter object names are not a good practice. Expand it out to be more descriptive.